core/src/main/java/org/mule/registry/MuleRegistryHelper.java

```diff
@@ -98,7 +102,7 @@
     /**
      * Transformers are registered on context start, then they are usually not unregistered
      */
-    private Collection<Transformer> transformers = new CopyOnWriteArrayList<Transformer>();
+    private Map<String, Transformer> transformers = new ConcurrentHashMap<String, Transformer>();
```

Comment:
Why not a set? From my understanding you just want to avoid duplicates.

Reply:
The Converter and Transformer hierarchy is not consistent. The abstract classes from which main converters and transformers inherit do not have the getName method (which is declared in the interface NameableObject). As a consequence of this, it not possible to create an equals method. I think refactoring that hierarchy violates retro-compatibility. The strings used for the key is similar to the one used for the registry (without the hyphen "_").